lib/timodule.js

```diff
@@ -739,13 +739,14 @@ async function detectPlatformSpecificNativeModuleInNPMPackage(directory, platfor
 	if (await fs.exists(manifestFile)) {
 		manifest = await readManifest(manifestFile);
 	}
-	const platformMetadadata = (json.titanium.platform && json.titanium.platform[platformName]) || {};
-	manifest.name = json.titanium.name || manifest.name;
-	manifest.minsdk = platformMetadadata.minsdk || manifest.minsdk;
-	manifest.apiversion = platformMetadadata.apiversion || parseInt(manifest.apiversion);
-	manifest.guid = json.titanium.guid || manifest.guid;
-	manifest.moduleid = json.titanium.moduleid || manifest.moduleid;
-	manifest.architectures = platformMetadadata.architectures || manifest.architectures.split(' ');
+	const platformMetadata = (json.titanium.platform && json.titanium.platform[platformName]) || {};
+	manifest.name = platformMetadata.name || json.titanium.name || manifest.name || json.name;
+	manifest.minsdk = platformMetadata.minsdk || json.titanium.minsdk || manifest.minsdk;
+	manifest.apiversion = platformMetadata.apiversion || json.titanium.apiversion || parseInt(manifest.apiversion);
+	manifest.guid = platformMetadata.guid || json.titanium.guid || manifest.guid;
+	manifest.moduleid = platformMetadata.moduleid || json.titanium.moduleid || manifest.moduleid || json.name;
+	manifest.architectures = platformMetadata.architectures || json.titanium.architectures || manifest.architectures.split(' ');
```

Comment:
@sgtcoolguy - I am assuming `architectures` is an array?  Should we check for string and split it in case they copy/paste architectures from manifest into package.json?  Or can it handle a string further down the pipeline?

Reply:
`manifest.architectures.split(' ');` should handle the copy/paste case from the manifest

Reply:
I was referring to platformMetadata.architectures

Reply:
I was referring to platformMetadata.architectures

Reply:
Ah damn, of course. I think we should keep it as an array there to settle for a single type in the package.json and not take over too much from the "old" manifest format.